concept/type_/mod.rs

```diff
@@ -77,8 +77,8 @@ pub trait TypeAPI<'a>: ConceptAPI<'a> + TypeVertexEncoding<'a> + Sized + Clone +
     ) -> Result<MaybeOwns<'m, Label<'static>>, ConceptReadError>;
 }
 
-pub trait KindAPI<'a>: TypeAPI<'a> + PrefixedTypeVertexEncoding<'a> {
-    type AnnotationType: Hash + Eq + From<Annotation>;
+pub trait KindAPI<'a>: TypeAPI<'a> {
+    type AnnotationType: Hash + Eq + Clone + From<Annotation> + Into<Annotation>;
```

Comment:
I do feel like we want a similar `KindAPI` for edges (relates, owns, plays), but I couldn't understand what a good term for it. So you'll see a ton of `EDGE: TypeEdgeEncoding<'static> + InterfaceImplementation<'static> + Clone` later

Reply:
The downside of `KindAPI::AnnotationType` is that our normal conversions of annotations return `Annotation` with a full list of possible `enum Annotation` instead of like `PlaysAnnotation`, etc. And it doesn't make the logic really smooth. But whatever